pubsub/topics/publish_settings.go

```diff
@@ -47,26 +44,20 @@ func publishWithSettings(w io.Writer, projectID, topicID string) error {
 		result := t.Publish(ctx, &pubsub.Message{
 			Data: []byte("Message " + strconv.Itoa(i)),
 		})
-
-		wg.Add(1)
-		go func(i int, res *pubsub.PublishResult) {
-			defer wg.Done()
-			// The Get method blocks until a server-generated ID or
-			// an error is returned for the published message.
-			id, err := res.Get(ctx)
-			if err != nil {
-				fmt.Fprintf(w, "Failed to publish: %v", err)
-				atomic.AddUint64(&totalErrors, 1)
-				return
-			}
-			fmt.Fprintf(w, "Published message %d; msg ID: %v\n", i, id)
-		}(i, result)
+		resultSlice = append(resultSlice, result)
 	}
-
-	wg.Wait()
-
-	if totalErrors > 0 {
-		return fmt.Errorf("%d of 10 messages did not publish successfully", totalErrors)
+	// The Get method blocks until a server-generated ID or
+	// an error is returned for the published message.
+	for i, res := range resultSlice {
+		id, err := res.Get(ctx)
+		if err != nil {
+			errors = append(errors, err)
+			fmt.Fprintf(w, "Failed to publish: %v", err)
+		}
+		fmt.Fprintf(w, "Published message %d; msg ID: %v\n", i, id)
+	}
+	if errors != nil {
+		return fmt.Errorf("%d of 10 messages did not publish successfully", len(errors))
```

Comment:
We should include an actual error in this result. Probably just the last one.

Reply:
Done